src/main/java/io/fabric8/maven/docker/wait/WaitUtil.java

```diff
@@ -36,17 +42,34 @@ public static long wait(int wait, Callable<Void> callable) throws ExecutionExcep
         return delta(now);
     }
 
-    public static long wait(int maxWait, WaitChecker ... checkers) throws WaitTimeoutException {
-        return wait(maxWait, Arrays.asList(checkers));
+    public static long wait(DockerAccess access, String containerId, int maxWait, WaitChecker ... checkers) throws WaitTimeoutException, NotRunningException {
+        return wait(access, containerId, maxWait, Arrays.asList(checkers));
     }
 
-    public static long wait(int maxWait, Iterable<WaitChecker> checkers) throws WaitTimeoutException {
+    public static long wait(DockerAccess access, String containerId, int maxWait, Iterable<WaitChecker> checkers) throws WaitTimeoutException, NotRunningException {
         long max = maxWait > 0 ? maxWait : DEFAULT_MAX_WAIT;
         long now = System.currentTimeMillis();
         try {
             do {
-                if (check(checkers)) {
-                    return delta(now);
+                try {
+                    if (!access.getContainer(containerId).isRunning()) {
+                        if (check(checkers)) {
```

Comment:
ok, I see the point. Let me pick this up and try whether I could keep the separation (you are probably right, it's not possible to use this is a check).